relay-event-normalization/src/normalize/processor.rs

```diff
@@ -736,20 +736,30 @@ fn normalize_performance_score(
                 if profile.score_components.iter().any(|c| {
                     !measurements.contains_key(c.measurement.as_str())
                         && c.weight.abs() >= f64::EPSILON
+                        && !c.optional.unwrap_or(false)
                 }) {
-                    // All measurements with a profile weight greater than 0 are required to exist
-                    // on the event. Skip calculating performance scores if a measurement with
-                    // weight is missing.
+                    // All non-optional measurements with a profile weight greater than 0 are
+                    // required to exist on the event. Skip calculating performance scores if
+                    // a measurement with weight is missing.
                     break;
                 }
                 let mut score_total = 0.0;
                 let mut weight_total = 0.0;
                 for component in &profile.score_components {
+                    // Skip optional components if they are not present on the event.
+                    if component.optional.is_some()
```

Comment:
Can be `unwrap_or(false)` as above

Reply:
Updated to just `component.optional` since I'm using a default value for optional now